src/widget/slider.rs

```diff
@@ -14,93 +14,104 @@
 
 //! A slider widget.
 
-use crate::widget::Widget;
-use crate::{BoxConstraints, HandlerCtx, Id, LayoutCtx, LayoutResult, MouseEvent, PaintCtx, Ui};
-
-use crate::kurbo::{Point, Rect, Size};
-use crate::piet::{Color, FillRule, RenderContext};
-
-const BOX_HEIGHT: f64 = 24.;
+use crate::kurbo::{Circle, Line, Point, Size, Vec2};
+use crate::piet::{Color, FillRule, RenderContext, StrokeStyle, LineCap};
+use crate::{
+    Action, BaseState, BoxConstraints, Env, Event, EventCtx, LayoutCtx, PaintCtx,
+    UpdateCtx, Widget,
+};
+
+const KNOB_WIDTH: f64 = 24.;
+const BACKGROUND_THICKNESS: f64 = 4.;
 const BACKGROUND_COLOR: Color = Color::rgb24(0x55_55_55);
 const SLIDER_COLOR: Color = Color::rgb24(0xf0_f0_ea);
 
+#[derive(Debug, Clone, Default)]
 pub struct Slider {
-    value: f64,
+    width: f64,
 }
 
-impl Slider {
-    pub fn new(initial_value: f64) -> Slider {
-        Slider {
-            value: initial_value,
-        }
-    }
 
-    pub fn ui(self, ctx: &mut Ui) -> Id {
-        ctx.add(self, &[])
-    }
-}
+impl Widget<f64> for Slider {
+    fn paint(&mut self, paint_ctx: &mut PaintCtx, base_state: &BaseState, data: &f64, _env: &Env) {
```

Comment:
I think for control widgets we should always have some visual indication of 'hover' and 'active' state. This doesn't need to be fancy now, but might be fun to paint the knob differently in these two cases?